### PR TITLE
Implement get_lp_token_price view instruction

### PR DESCRIPTION
**Pull request by the-orex**

### CODE REVIEW

File: programs/perpetuals/src/instructions/get_lp_token_price.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    let lp_supply = ctx.accounts.lp_token_mint.supply;
+
+    msg!("lp_supply: {}", lp_supply);
+
+    if lp_supply.is_zero() {
+        return Ok(0);
+    }
+
+    let ui_price_usd = checked_float_div(checked_as_f64(aum_usd)?, checked_as_f64(lp_supply)?)?;
```
Comment (Adrena-Corto) [inline programs/perpetuals/src/instructions/get_lp_token_price.rs]: Other use of this functions us a implicit cast from u64 to f64 in the Askibin code, but that's theoretically unsafe and should use the checked_as_f64 (f64 can only store 53bits)

File: programs/perpetuals/src/math.rs [hunk 2]
```diff
@@ -353,6 +353,19 @@ where
     }
 }
 
+pub fn checked_as_f64<T>(arg: T) -> Result<f64>
```
Comment (Adrena-Corto) [inline programs/perpetuals/src/math.rs:356]: OK clean. I think in another PR we might want to use that for any u64 -> f64 casts throughout the codebase, weird their audit didn't pick it up?

File: programs/perpetuals/tests/native/tests_suite/lp_token/lp_token_price.rs [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+                },
+                utils::SetupCustodyWithLiquidityParams {
+                    setup_custody_params: utils::SetupCustodyParams {
+                        mint: eth_mint,
+                        decimals: ETH_DECIMALS,
+                        is_stable: false,
+                        target_ratio: utils::ratio_from_percentage(50.0),
+                        min_ratio: utils::ratio_from_percentage(0.0),
+                        max_ratio: utils::ratio_from_percentage(100.0),
+                        initial_price: utils::scale(1_500, ETH_DECIMALS),
```
Comment (Adrena-Corto) [inline programs/perpetuals/tests/native/tests_suite/lp_token/lp_token_price.rs:122]: unrelated: need to make a PR on our end once I finish staking/multiplier to realign these

File: programs/perpetuals/tests/native/utils/utils.rs [hunk 4]
```diff
@@ -201,6 +202,52 @@ pub async fn create_and_fund_multiple_accounts(
     keypairs
 }
 
+pub async fn create_and_simulate_perpetuals_view_ix<T: InstructionData, U: BorshDeserialize>(
```
Comment (Adrena-Corto) [inline programs/perpetuals/tests/native/utils/utils.rs:205]: 👏

**Review by Adrena-Corto — DISMISSED** [review]
Nits only. LGTM

**Comment by the-orex** [comment]
Opened on main repo

**Pull request closed without merging**